File: Documentation/5.4/Samples/python/ClientApi/Session/StoringEntities.py
from typing import Optional

from ravendb.infrastructure.orders import Employee

from examples_base import ExamplesBase


class StoringEntities(ExamplesBase):
    def setUp(self):
        super().setUp()

    class IFoo:
        # region store_entities_1
        def store(self, entity: object, key: Optional[str] = None, change_vector: Optional[str] = None) -> None:
            ...

        # endregion

    def test_storing_entities(self):
        with self.embedded_server.get_document_store("StoringEntities") as store:
            with store.open_session() as session:
                # region store_entities_5
                session.store(Employee(first_name="John", last_name="Doe"))

                # send all pending operations to server, in this case only 'Put' operation
                session.save_changes()
                # endregion

File: Documentation/5.4/Samples/php/ClientApi/session/StoringEntities.php
<?php

namespace RavenDB\Samples\ClientApi\Session;

use RavenDB\Samples\Infrastructure\DocumentStoreHolder;
use RavenDB\Samples\Infrastructure\Orders\Employee;

interface FooInterface {
    # region store_entities_1
    public function store(?object $entity): void;
    # endregion

    # region store_entities_2
    public function store(?object $entity, ?string $id): void;
    # endregion

    # region store_entities_3
    public function store(?object $entity, ?string $id, ?string $changeVector): void;
    # endregion

}


class StoringEntities
{

    public function storeEntities(): void
    {
        $store = DocumentStoreHolder::getStore();
        try {
            $session = $store->openSession();
            try {
                # region store_entities_5
                $employee = new Employee();
                $employee->setFirstName("John");
                $employee->setLastName("Doe");

                // generate Id automatically
                $session->store($employee);

                // send all pending operations to server, in this case only `Put` operation
                $session->saveChanges();
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
